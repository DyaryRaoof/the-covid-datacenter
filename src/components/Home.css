.top-section,
.top-section-content {
  height: 200px;
  background-color: #fa4f91;
}

.stats {
  background-color: #ec4c8a;
  margin-left: 10px;
}

.country-grid {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.single-grid-item {
  height: 150px;
}

.light-background {
  background-color: #e7578e;
}

.grid-item-navlink {
  color: white;
  border: none;
}

.dark-background {
  background-color: #d6487e;
}
