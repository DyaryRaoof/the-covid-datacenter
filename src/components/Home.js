const Home = () => <div>Hello from home</div>;

export default Home;
