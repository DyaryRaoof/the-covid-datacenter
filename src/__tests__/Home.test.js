import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../components/Home';
import store from '../redux/configureStore';

describe('test Home component', () => {
  test('should contain a image emelement', () => {
    render(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    return expect(screen.getByTestId('image')).toBeInTheDocument;
  });
});

describe('test with jest snapshot', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
