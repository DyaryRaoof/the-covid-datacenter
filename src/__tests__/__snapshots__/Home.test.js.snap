// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test with jest snapshot should render correctly 1`] = `
<main>
  <section
    className="top-section"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <img
          alt="Europe map"
          className="map opacity-25 ms-4"
          data-testid="image"
          src="united-states-map.png"
        />
      </div>
      <div
        className="col"
      >
        <div
          className=" top-section-content d-flex flex-column justify-content-center align-items-start"
        >
          <h1>
            United States
          </h1>
          <span>
            0
          </span>
        </div>
      </div>
    </div>
  </section>
  <span
    className="stats"
  >
    TODAYS CONFIRMED CASES
  </span>
  <section>
    <div
      className="country-grid"
    />
  </section>
</main>
`;
